### PR TITLE
[15.0][MIG] sale_order_priority

### PR DESCRIPTION
**Pull request by JuanyDForgeflow**
This module adds the field *Priority* in sale order lines and sale orders:
priority of the sale order is computed as the maximum of the priorities of its
lines, setting the priority in the order sets the priority of all its lines
accordingly.

When a picking is created as a result of sale order confirmation,
the created procurement inherits the priority of the order,
then the stock moves and the picking inherit the procurement's priority.

### CODE REVIEW

**Review by MiquelRForgeFlow — APPROVED** [review]

File: sale_order_priority/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from odoo import models
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
+
+    def action_assign(self):
+        res = super(Picking, self).action_assign()
```
Comment (cesar-tecnativa) [inline sale_order_priority/models/stock_picking.py]: super().action_assign()

**Review by cesar-tecnativa — CHANGES_REQUESTED** [review]

File: sale_order_priority/models/stock_picking.py (hunk 1 above)
Comment (JuanyDForgeflow) [reply sale_order_priority/models/stock_picking.py]: pre-commit fails if I don't return the super

Comment (MiquelRForgeFlow) [reply sale_order_priority/models/stock_picking.py]: the suggestion was not to delete the `res=`  + `return`. please, put this again. the suggestion was to delete the content inside super()

Comment (JuanyDForgeflow) [reply sale_order_priority/models/stock_picking.py]: @MiquelRForgeFlow @cesar-tecnativa Done! :)

File: sale_order_priority/models/stock_picking.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from odoo import models
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
+
+    def action_assign(self):
+        res = super().action_assign()
+        picking_priority = self.env.context.get("sale_priority")
+        self.priority = picking_priority if picking_priority else 0
```
Comment (GuillemCForgeFlow) [inline sale_order_priority/models/stock_picking.py]:
```suggestion
    def action_confirm(self):
        res = super().action_confirm()
        picking_priority = self.env.context.get("sale_priority", False)
        self.write({"priority": picking_priority or 0})
```

Comment (GuillemCForgeFlow) [reply sale_order_priority/models/stock_picking.py]: From what I've tested, the action_assign method doesn't seem to do the trick anymore. With the action_assing the module was not working, the priority of the order was not passed to the picking. Applying the changes,  it should work again! :+1:

**Review by GuillemCForgeFlow — CHANGES_REQUESTED** [review]

**Review by GuillemCForgeFlow — APPROVED** [review]
LGTM! :+1:

File: sale_order_priority/models/stock_picking.py (hunk 2 above)
Comment (JuanyDForgeflow) [reply sale_order_priority/models/stock_picking.py]: @GuillemCForgeFlow Thanks! it's working now

**Comment by AaronHForgeFlow** [comment]
@cesar-tecnativa the comments are attended. Can you please update the review? Thanks

**Review by AaronHForgeFlow — APPROVED** [review]
Code LG!

**Review by manuelregidor — APPROVED** [review]
Functional and technical review. LGTM

**Review by ValentinVinagre — APPROVED** [review]
LGTM 👍🏻

**Comment by manuelregidor** [comment]
@HaraldPanten

**Comment by ValentinVinagre** [comment]
@HaraldPanten

**Review by HaraldPanten — APPROVED** [review]

**Pull request merged**